src/fluree/db/query/fql/parse.cljc

```diff
@@ -328,7 +328,8 @@
   [[s-pat p-pat o-pat] db context]
   (let [s (parse-subject-pattern s-pat context)
         p (parse-predicate-pattern p-pat db context)]
-    (if (= const/$rdf:type (::where/val p))
+    (if (and (= const/$rdf:type (::where/val p))
+             (not (syntax/variable? o-pat)))
```

Comment:
Queries with an iri predicate and an object var will fail in exactly the same way that this has been, eg
```clj
'{:select   [?id]
  :where    [[?s :rdf/type :ex/User]
             [?s :id ?id]]} ;;<-- this clause in particular
```
will throw the same `expand` error that this was previously.

Is there a reason to not support that pattern? It'd be trivial to apply the same fix for that too.

Reply:
If we shouldn't support it for some reason, we could at least throw a better error message.